cpp/src/arrow/compute/kernels/scalar_cast_temporal.cc

```diff
@@ -519,6 +521,16 @@ std::shared_ptr<CastFunction> GetDurationCast() {
   // Between durations
   AddCrossUnitCast<DurationType>(func.get());
 
+  // time32/64 -> duration
```

Comment:
IMO, it doesn't make sense to cast time to duration.

Reply:
Indeed. Removed.